spec/duration.html

```diff
@@ -434,31 +431,28 @@ <h1>Temporal.Duration.prototype.round ( _roundTo_ )</h1>
         1. If _maximum_ is not *undefined*, perform ? ValidateTemporalRoundingIncrement(_roundingIncrement_, _maximum_, *false*).
         1. If _roundingIncrement_ > 1, and _largestUnit_ is not _smallestUnit_, and IsCalendarUnit(_smallestUnit_) is *true* or _smallestUnit_ is *"day"*, then
           1. Throw a *RangeError* exception.
-        1. Let _norm_ be NormalizeTimeDuration(_duration_.[[Hours]], _duration_.[[Minutes]], _duration_.[[Seconds]], _duration_.[[Milliseconds]], _duration_.[[Microseconds]], _duration_.[[Nanoseconds]]).
         1. If _zonedRelativeTo_ is not *undefined*, then
+          1. Let _normalizedDuration_ be NormalizeDuration(_duration_).
           1. Let _timeZone_ be _zonedRelativeTo_.[[TimeZone]].
           1. Let _calendar_ be _zonedRelativeTo_.[[Calendar]].
           1. Let _relativeEpochNs_ be _zonedRelativeTo_.[[Nanoseconds]].
-          1. Let _normalizedDuration_ be ! CreateNormalizedDurationRecord(_duration_.[[Years]], _duration_.[[Months]], _duration_.[[Weeks]], _duration_.[[Days]], _norm_).
           1. Let _targetEpochNs_ be ? AddZonedDateTime(_relativeEpochNs_, _timeZone_, _calendar_, _normalizedDuration_).
           1. Let _roundRecord_ be ? DifferenceZonedDateTimeWithRounding(_relativeEpochNs_, _targetEpochNs_, _timeZone_, _calendar_, _largestUnit_, _roundingIncrement_, _smallestUnit_, _roundingMode_).
-          1. Let _roundResult_ be _roundRecord_.[[DurationRecord]].
-        1. Else if _plainRelativeTo_ is not *undefined*, then
-          1. Let _targetTime_ be AddTime(0, 0, 0, 0, 0, 0, _norm_).
+          1. Return _roundRecord_.[[Duration]].
+        1. If _plainRelativeTo_ is not *undefined*, then
+          1. Let _normalizedDuration_ be NormalizeDurationWith24HourDays(_duration_).
+          1. Let _targetTime_ be AddTime(0, 0, 0, 0, 0, 0, _normalizedDuration_.[[NormalizedTime]]).
           1. Let _isoRelativeToDate_ be TemporalObjectToISODateRecord(_plainRelativeTo_).
           1. Let _calendar_ be _plainRelativeTo_.[[Calendar]].
-          1. Let _dateDuration_ be CreateDurationRecord(_duration_.[[Years]], _duration_.[[Months]], _duration_.[[Weeks]], _duration_.[[Days]] + _targetTime_.[[Days]], 0, 0, 0, 0, 0, 0).
+          1. Let _dateDuration_ be ? AdjustDateDurationRecord(_normalizedDuration_.[[Date]], _targetTime_.[[Days]]).
           1. Let _targetDate_ be ? CalendarDateAdd(_calendar_, _isoRelativeToDate_, _dateDuration_, *"constrain"*).
```

Comment:
Oh, this should be `!` (and the same for your other comment below.) I'll fix this in a follow up PR. (Proof: _targetTime_.[[Days]] cannot exceed the original value of _duration_.[[Days]] because its absolute value can only be equal or smaller, coming out of AddTime.)